metabase/metabase
# make the newsletter signup a checkbox instead of a button

Part of  [[Epic] Reduce the number of choices in onboarding flow](https://github.com/metabase/metabase/issues/46741)

# Description
This changes the newsletter subscription to be a checkbox instead of a button.
The technical details are explained in the [tech plan](https://www.notion.so/metabase/Tech-Reduce-the-number-of-choices-in-onboarding-flow-9def6e76d62d4c4f86b269191324ce69?pvs=4#e094a53229214ebf91f2da3c6a5a400b) but the tldr is that we use navigator.sendBeacon to make sure the request is not lost when navigating away from the page.

# How to verify
This is a bit tricky, to do a full test I had a mock backend logging what it received, i had it listening on a custom domain over https to make sure that the request successfully reached the server even with cors/https etc. (see video under the demo).
If you want to try it, you'll have to add the custom domain you're using in the [security.clj file](https://github.com/metabase/metabase/blob/60aa323343641d16c0b3fe21fd5df714a660af77/src/metabase/server/middleware/security.clj#L99-L103)
It seems it's working ok but if you have any issue or concern please raise it as I don't want to break the newsletter signup 🙃 

# Demo
<img width="1051" alt="image" src="https://github.com/user-attachments/assets/6b3c4a05-cfcc-40e3-8499-33d6dc331c83">

https://github.com/user-attachments/assets/b4511c67-f664-4efd-97eb-2a8cf8c23ef9
